model/task_lifecycle.go

```diff
@@ -2468,10 +2477,19 @@ func checkResetSingleHostTaskGroup(ctx context.Context, t *task.Task, caller str
 		return errors.Errorf("no tasks in task group '%s' for task '%s'", t.TaskGroup, t.Id)
 	}
 	shouldReset := false
+	isAutoRestart := false
 	for _, tgTask := range tasks {
 		if tgTask.ResetWhenFinished {
 			shouldReset = true
 		}
+		if tgTask.IsAutomaticRestart {
+			// Do not restart the task group if it contains a task marked for auto restart
+			// that has since been aborted
+			if tgTask.Aborted {
+				return nil
+			}
+			isAutoRestart = true
```

Comment:
can we just set requester here to AutoReset? Then we don't need the variable

Reply:
True! fixed.